src/VuData.cpp

```diff
@@ -42,22 +42,24 @@ using namespace vuplus;
 using namespace ADDON;
 using namespace P8PLATFORM;
 
-Vu::Vu() 
+Vu::Vu(const Settings &settings) 
+  : m_settings(settings)
 {
   m_bIsConnected = false;
   m_strServerName = "Vu";
   std::string strURL = "";
+  entryExtractor = std::unique_ptr<EpgEntryExtractor>(new EpgEntryExtractor(m_settings));
```

Comment:
Hm, a wild mix of init list, member init in body and local vars here...

Reply:
Is there is way to initialise entryExtractor in the header file? Was stumped trying to figure that out.

Reply:
doing complex init in the header is no good style; i would keep it in the cpp